### PR TITLE
tidy oeo imports

### PR DESCRIPTION
**Pull request by stap-m**
ToDos here
- [x] comment on txt files with import declarations: iao, omo
- [x] include `may be identical to` to iao import

ToDo in oeo repo
- [x] add new iao-extracted.owl from this repo
- [x] remove omo import
- [x] oeo-import-edits: --> see see https://github.com/OpenEnergyPlatform/ontology/pull/1754
    - [x] domain of `is about`
    - [x] parent class of `information content entity`

Next steps

- [x] add aditional classes to iao imports

### CODE REVIEW

**Comment by stap-m** [comment]
Seems like all annotation properties and entities we import from omo, apart from `may be identical to`, are also imported from iao (implicitly or explicitly), hence we have them twice. 
`may be identical to` can also be imported from iao, instead, which would make the omo import obsolete.

File: oeo-imports/iao/extract-iao-module.sh [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-# Extract domain of IAO_0000136
-robot merge --input iao-full-download.owl filter --term http://purl.obolibrary.org/obo/IAO_0000136 --term rdfs:domain --term http://purl.obolibrary.org/obo/IAO_0000030 --select self  --axioms all --signature false   --output iao-extracted-domain.owl
-# Extracted module
-robot merge  --input iao-module-temp.owl --input iao-extracted-domain.owl --output ../../ontology/imports/iao-extracted.owl
-# Remove subproperty axioms
-robot remove --input  ../../ontology/imports/iao-extracted.owl --axioms subproperty --output  ../../ontology/imports/iao-extracted.owl
-# Annotates the output with a commentary to the origin of the content
-robot annotate --input ../../ontology/imports/iao-extracted.owl --annotation rdfs:comment "This file contains externally imported content from the Information Artifact Ontology (IAO) for import into the Open Energy Ontology (OEO). It is automatically extracted using ROBOT." --output ../../ontology/imports/iao-extracted.owl
+# Extract the terms we want with hierarchy until the upper term "informatio content entity" (IAO_0000030).
```
Comment (stap-m) [inline oeo-imports/iao/extract-iao-module.sh]:
```suggestion
# Extract the terms we want with hierarchy until the upper term "information content entity" (IAO_0000030). 
```

File: oeo-imports/iao/extract-iao-module.sh [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-# Remove subproperty axioms
-robot remove --input  ../../ontology/imports/iao-extracted.owl --axioms subproperty --output  ../../ontology/imports/iao-extracted.owl
-# Annotates the output with a commentary to the origin of the content
-robot annotate --input ../../ontology/imports/iao-extracted.owl --annotation rdfs:comment "This file contains externally imported content from the Information Artifact Ontology (IAO) for import into the Open Energy Ontology (OEO). It is automatically extracted using ROBOT." --output ../../ontology/imports/iao-extracted.owl
+# Extract the terms we want with hierarchy until the upper term "information content entity" (IAO_0000030). 
+# Warning: This removes the domain of IAO_0000136 -- tbd in oeo-import-edits
+# Classification:  tbd in oeo-import-edits
+robot merge --input C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-full-download.owl extract --method MIREOT --lower-terms C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-w-hierarchy.txt --intermediates all --upper-term http://purl.obolibrary.org/obo/IAO_0000030 --output C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl
```
Comment (stap-m) [inline oeo-imports/iao/extract-iao-module.sh]:
```suggestion
robot merge --input iao-full-download.owl extract --method MIREOT --lower-terms iao-w-hierarchy.txt --intermediates all --upper-term http://purl.obolibrary.org/obo/IAO_0000030 --output iao-extracted.owl
```

File: oeo-imports/iao/extract-iao-module.sh [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
+# Annotates the output module with a commentary to the origin of the content
+robot annotate --input C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl --annotation rdfs:comment "This file contains externally imported content from the Information Artifact Ontology (IAO) for import into the Open Energy Ontology (OEO). It is automatically extracted using ROBOT." --output C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl
```
Comment (stap-m) [inline oeo-imports/iao/extract-iao-module.sh]:
```suggestion
robot annotate --input iao-extracted.owl --annotation rdfs:comment "This file contains externally imported content from the Information Artifact Ontology (IAO) for import into the Open Energy Ontology (OEO). It is automatically extracted using ROBOT." --output iao-extracted.owl
```

File: oeo-imports/iao/extract-iao-module.sh [hunk 4]
```diff
@@ -1,22 +1,14 @@
...
 # Annotates each axiom with the ontology IRI, using prov:wasDerivedFrom
-robot annotate --input ../../ontology/imports/iao-extracted.owl --annotate-derived-from true --annotate-defined-by true --output ../../ontology/imports/iao-extracted.owl
+robot annotate --input C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl --annotate-derived-from true --annotate-defined-by true --output C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl
```
Comment (stap-m) [inline oeo-imports/iao/extract-iao-module.sh]:
```suggestion
robot annotate --input iao-extracted.owl --annotate-derived-from true --annotate-defined-by true --output iao-extracted.owl
```

File: oeo-imports/iao/extract-iao-module.sh [hunk 5]
```diff
@@ -1,22 +1,14 @@
...
 # Annotate with new ontology information
-robot annotate --input ../../ontology/imports/iao-extracted.owl  --ontology-iri http://openenergy-platform.org/ontology/oeo/imports/iao-extracted.owl --version-iri http://openenergy-platform.org/ontology/oeo/dev/imports/iao-extracted.owl --output ../../ontology/imports/iao-extracted.owl
+robot annotate --input C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl  --ontology-iri http://openenergy-platform.org/ontology/oeo/imports/iao-extracted.owl --version-iri http://openenergy-platform.org/ontology/oeo/dev/imports/iao-extracted.owl --output C:/Users/stappel/Documents/Ontologies/test_oeo_imports/iao-extracted.owl
```
Comment (stap-m) [inline oeo-imports/iao/extract-iao-module.sh]:
```suggestion
robot annotate --input iao-extracted.owl  --ontology-iri http://openenergy-platform.org/ontology/oeo/imports/iao-extracted.owl --version-iri http://openenergy-platform.org/ontology/oeo/dev/imports/iao-extracted.owl --output iao-extracted.owl
```

**Review by stap-m — COMMENTED** [review]
remove local paths

**Comment by stap-m** [comment]
@h-spinde the new iao-extracted looks fine for me.
As a next step, please copy it to the oeo repo and perform the following steps in a new PR. Let me know, if you have questions.

> ToDo in oeo repo
>  - [x] add new iao-extracted.owl from this repo
> - [x] remove omo import
> - [x] oeo-import-edits:
>     - [x] domain of `is about`
>     - [x] parent class of `information content entity`

EDIT: see https://github.com/OpenEnergyPlatform/ontology/pull/1754

**Comment by stap-m** [comment]
@h-spinde please run the scipt again based on the newest version of iao-w-hierarchy.sh in this PR and push the result here. Please let me know once its done.
Does ist cause problems that there is an OBI class involved which IAO seems to import?

**Comment by h-spinde** [comment]
I have run the new script and uploaded it. There shouldn't be any problems caused by the imported OBI class, especially since it isn't actually used anywhere. Though I am admittedly confused by how it is getting imported in the first place, since it's neither in iao-w-hierarchy.txt, nor relevant to any other class being imported.

**Comment by stap-m** [comment]
The OBI class was requested newly and wasn't imported before.

**Comment by stap-m** [comment]
@h-spinde can you run the script with the updated iao-w-hierarchy on short notice, i.e. today or tomottow?

EDIT: I managed if myself now.

**Comment by areleu** [comment]
Tried doing the axiom cleanup using `robot merge -i iao-extracted.owl repair --merge-axiom-annotations true -o iao-extracted.owl` from: https://github.com/ontodev/robot/issues/1099#issuecomment-1476875014 
But it breaks the ontology so we may need other solution, since this import is reproducible I think this is good to merge

**Review by areleu — APPROVED** [review]
See my latest comment

**Pull request merged**